### ISSUE TITLE
DataMapper Map accepting values from methods

### ISSUE THREAD

**Issue by Brannos1970** [MarimerLLC/cslaforum#775]
I was wondering if the Datamapper Map feature can accept something like:

`Map(CustomerStatus, ParseEnum<StatusEnums>(CustomerStatus)`

At the moment I exclude fields that need some type of conversion. Which is not a problem but it would simplify some things.

I may be overthinking this or overlooking something.

**Comment by rockfordlhotka** [comment]
Is `CustomerStatus` a property or an object?

DataMapper maps entire objects, not individual properties.

**Comment by Brannos1970** [comment]
It is a property in the object. I was thinking this could be part of the custom Map that we could build.

**Comment by rockfordlhotka** [comment]
It sounds like what you are describing is a value converter - an optional function that can be invoked during a property level mapping operation to convert a value. XAML data binding has this sort of thing built in, for example, via an `IValueConverter` interface.

Yes, that is the sort of thing that could be an enhancement to DataMapper, in the case that a custom map is provided.

Is that something you are interested in implementing?

**Comment by Brannos1970** [comment]
Yes. I can see how I would use it and that may be a good starting point but I am sure others may have additional requirements that I have not considered yet. As soon as I can make some time I will work on this.

**Comment by ajj7060** [comment]
@rockfordlhotka Sorry for straying a little off-topic, but does DataMapper do anything with the clsa objects metastate, or does it just copy values from one object to another?  I've never used DataMapper, but it seems like maybe it could be done using https://github.com/AutoMapper/AutoMapper.  (Which has value converter support).  Just thinking maybe there's one less component to maintain in Csla.
